s3transfer/download.py

```diff
@@ -192,6 +203,9 @@ def __init__(self, osutil, transfer_coordinator, io_executor,
     def is_compatible(cls, download_target):
         return hasattr(download_target, 'write')
 
+    def buffers_body_in_memory(self):
```

Comment:
Can you talk about the thought/motivation behind adding a new method to the manager?  This seems to be leaking implementation details (i.e. why does it matter to a caller how the manager decides to manager the internals of the body?)

Given the point of this method is solely (for now) to figure out the appropriate tags for the task, how is this approach better than just directly asking the download manager about any tags to associate with the task?  e.g `download_manager.get_tags()` or `download_manager.get_task_tags()`, etc.